Ginger/Ginger/SolutionWindows/TreeViewItems/BusinessFlowTreeItem.cs

```diff
@@ -139,7 +141,36 @@ private void ActionsConversionHandler(object sender, System.Windows.RoutedEventA
 
             WizardWindow.ShowWizard(new ActionsConversionWizard(ActionsConversionWizard.eActionConversionType.MultipleBusinessFlow, new Context(), lst), 900, 700, true);
         }
-        
+
+        /// <summary>
+        /// This method helps to execute the funcationality of removeing the legacy actions from the businessflow
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void LegacyActionsRemoveHandler(object sender, System.Windows.RoutedEventArgs e)
+        {
+            ObservableList<BusinessFlowToConvert> lstBFToConvert = new ObservableList<BusinessFlowToConvert>();
+            if (((ITreeViewItem)this).NodeObject().GetType().Equals(typeof(GingerCore.BusinessFlow)))
```

Comment:
1- why you have this IF-Else? code in IF should be here (BusinessFlowTreeItem) and else code should be in BusinessFlowFolderTreeITem, they can't be both so no need IF-Else and duplicated code

2- you dont need to get the tree item object like ((ITreeViewItem)this).NodeObject(), you have "mBusinessFlow", and "mBusinessFlowFolder" (in folderTreeItem) for that

Reply:
Changed